src/assets/scripts/util/autofit_images.js

```diff
@@ -1,241 +1,161 @@
-}
-
-function getPosition(element) {
-    var yPosition = 0;
-
-    while(element) {
-        yPosition += (element.offsetTop - element.scrollTop + element.clientTop);
-        element = element.offsetParent;
-    }
-
-    return yPosition;
-}
-
-  function animateImageIfjQuery (element, source, backgroundImage) {
-    element.alreadyLoaded = true;
-    if (typeof element != 'undefined' && typeof source != 'undefined' && typeof backgroundImage != 'undefined') {
-      if ((!element.src || element.src.length === 0) && !backgroundImage) {
-		    element.src = source;
-
-        var originalElementOpacity = element.style.opacity;
-        if (!originalElementOpacity || originalElementOpacity == 0) {
-          originalElementOpacity = 1;
+import $ from 'jquery';
+ 
+
+var autoFitImages = {
+  el: {
+    newImg: new Image
+  },
+  cacheSelectors: function () {
+    this.el.backgroundString = 'backgroundimagesrc';
+    this.el.imageString = 'imagebasesrc';
+
+  },
+  init: function(self = this) {
+    self.cacheSelectors();
+    $(window).resize(function () {
+      self.loadImagesByWidth(self.el.imageString, self.el.backgroundString);
+    });
+    $(window).scroll(function () {
+      self.loadIfVisible();
+    });
+    self.loadImagesByWidth(self.el.imageString, self.el.backgroundString);
+    self.loadIfNotVisible(0);    
+
+  },
+  getClosestWidth: function(element, baseNode) {
+      var elementWidth = element.offsetWidth;
+      var baseWidthCheck = 50;
+      var widthMaximum = 4800;
+      var diffFromWidth = widthMaximum;
+      var chosenNode = "";
+      for (var i = baseWidthCheck; i < widthMaximum; i+=50) {
+        if (Math.abs(elementWidth - i) < diffFromWidth && element && element.attributes && element.attributes[baseNode+i] && element.attributes[baseNode+i].nodeValue && element.attributes[baseNode+i].nodeValue.length > 1) {
+          diffFromWidth = elementWidth - i;
+          chosenNode = baseNode+i;
+        }
+      }
+      return chosenNode;
+  },
+  getPosition: function(element) {
+        var yPosition = 0;
+
+        while(element) {
+            yPosition += (element.offsetTop - element.scrollTop + element.clientTop);
+            element = element.offsetParent;
         }
-        element.style.opacity = 0;
 
-        element.onload = function() {
-          element.style.opacity = 0;
-          if (window.jQuery) {
+        return yPosition; 
+  },
+  showImage: function(element, source, backgroundImage) {
+        element.alreadyLoaded = true;
+        if (typeof element != 'undefined' && typeof source != 'undefined' && typeof backgroundImage != 'undefined') {
+          if ((!element.src || element.src.length === 0) && !backgroundImage) {
+            element.src = source;
+
+            var originalElementOpacity = $(element).css('opacity');
+            if (!originalElementOpacity || originalElementOpacity == 0) {
+              originalElementOpacity = 1;
+            }
+            $(element).css('opacity', 0);

```

Comment:
pull element into a var like $element up top so it's not being rewrapped into a jquery element multiple times.